src/dataset.ts

```diff
@@ -143,28 +143,31 @@ export interface Environment {
     cutoff: number;
 }
 
+/** Arbitrary javascript object, to be validated */
+export type JsObject = Record<string, unknown>;
+
 /** @hidden
  * Check that the given object, potentially comming from javascript, has all
  * required properties to be a dataset.
  */
-export function checkDataset(o: any) {
-    if (!('meta' in o && typeof o.meta === 'object')) {
+export function validateDataset(o: JsObject) {
+    if (!('meta' in o && typeof o.meta === 'object' && o.meta !== null)) {
         throw Error('missing "meta" key in dataset');
```

Comment:
```suggestion
        throw Error('Missing "meta" key in dataset containing the metadata.');
```

Reply:
In general I think the errors could be more descriptive of what they're flagging.

Reply:
I tried to improve the messages, although in this specific case more specific messages will be given in the `checkMetadata` function